src/session.rs

```diff
@@ -62,12 +62,19 @@ impl Session {
             let input = string.as_bytes();
             percent_encoding::lossy_utf8_percent_decode(input)
         }
+        fn clean_vhost(string: String) -> String {
```

Comment:
Not sure why we need String -> String here (which requires extra allocations).
&str would be fine:

```rust
fn clean_vhost(string: &str) -> &str {
    if string.chars().next() == Some('/') {
      decode(&string[1..])
    } else {
      decode(&string[..])
    }
}
```